test/Splitter.t.sol

```diff
@@ -0,0 +1,408 @@
+            Splitter.kick.selector,
+            Splitter.cage.selector
+        ]);
+    }
+
+    function testFileUint() public {
+        checkFileUint(address(splitter), "Splitter", ["burn"]);
+    }
+
+    function testFileAddress() public {
+        checkFileAddress(address(splitter), "Splitter", ["flapper"]);
+    }
+
+    function testVatCanAfterFile() public {
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf1));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 1);
+        assertEq(vat.can(address(splitter), address(0xf2)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf2));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+        assertEq(vat.can(address(splitter), address(0xf2)), 1);
+    }
+
+    function testKick() public {
+        doKick();
+    }
+
+    function testKickBurnOnly() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop());
+
+        vm.prank(PAUSE_PROXY); splitter.file("burn", WAD);
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        doKick();
+    }
+
+    function testKickAfterHop() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop());
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        doKick();
+    }
+
+    function testKickBeforeHop() public {
+        doKick();
+        vm.warp(block.timestamp + flapper.hop() - 1 seconds);
+
+        // make sure the slippage of the first kick doesn't block us
+        uint256 _marginalWant = marginalWant(MKR, address(medianizer));
+        vm.prank(PAUSE_PROXY); flapper.file("want", _marginalWant * 99 / 100);
+        vm.expectRevert("FlapperUniV2SwapOnly/kicked-too-soon");
+        vow.flap();
+    }
+
+    function testKickFlapperNotSet() public {
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0));
+        vm.expectRevert(bytes(""));
+        vow.flap();
+    }
+
+    function testChangeRewardDuration() public {
+        assertEq(flapper.hop(), farm.rewardsDuration());
+
+        doKick(); // sets periodFinish to now + rewardsDuration
+
+        vm.expectRevert("Previous rewards period must be complete before changing the duration for the new period");
+        vm.prank(PAUSE_PROXY); farm.setRewardsDuration(666 minutes); 
+
+        vm.prank(PAUSE_PROXY); splitter.file("burn", WAD);
+
+        vm.warp(block.timestamp + farm.rewardsDuration() + 1);
+        doKick();
+
+        vm.prank(PAUSE_PROXY); farm.setRewardsDuration(666 minutes); // should not revert
```

Comment:
I think we should add:

-A test for increasing the rewards duration and hop while there is a farming program. If I'm not mistaking we will have one intermediate period with a lower reward rate (between the setDuration/file(hop) and the end of the new hop), then should start the new cadence with the new reward rate.

-A test for increasing the rewards duration and hop while there is a farming program. Similarly we should have one intermediate period with a higher reward rate.

Reply:
Yes, you're right, done.

Reply:
nicely done